litestar/handlers/websocket_handlers.py

```diff
@@ -246,7 +281,9 @@ def _update_listener_fn_signature(listener_fn: Callable, callback_signature: ins
         listener_fn.__annotations__ = {p.name: p.annotation for p in new_signature.parameters.values()}
 
     def __call__(self, listener_callback: AnyCallable) -> websocket_listener:
-        listener_callback_signature = ParsedSignature.from_fn(listener_callback, self.resolve_signature_namespace())
```

Comment:
There is an issue with calling `resolve_signature_namespace()` inside `__call__()`, which is that the hierarchy of the handler hasn't been established yet.

Reply:
How could we fix that? We do need that information there and I don't immediately see a way to delay this.

Reply:
I don't know that there _is_ a way to have the hierarchy resolvable from within `__call__()`.. for the same reason I pushed anything that accesses the `parsed_fn_signature` property into `on_registration()`.

Reply:
Have addressed this in the latest commit.